File: frontend/types/bundle.ts
export interface BundleItem {
  product_id: string;
  product_name: string;
  quantity: number;
}

export interface Bundle {
  bundle_id: string;
  name: string;
  description: string;
  price: number;
  active: boolean;
  items: BundleItem[];
  individual_total: number | null;
  savings: number | null;
}

export interface BundleItemInput {
  product_id: string;
  quantity: number;
}

export interface BundleCreateRequest {
  bundle_id: string;
  name: string;
  description?: string;
  price: number;
  items: BundleItemInput[];
}

export interface BundleUpdateRequest {
  name?: string;
  description?: string;
  price?: number;
  active?: boolean;
  items?: BundleItemInput[];
}

File: backend/app/schemas/bundle.py
from pydantic import BaseModel


class BundleItemResponse(BaseModel):
    product_id: str
    product_name: str
    quantity: int


class BundleResponse(BaseModel):
    bundle_id: str
    name: str
    description: str
    price: float
    active: bool
    items: list[BundleItemResponse]
    individual_total: float | None = None
    savings: float | None = None


class BundleItemInput(BaseModel):
    product_id: str
    quantity: int


class BundleCreateRequest(BaseModel):
    bundle_id: str
    name: str
    description: str = ""
    price: float
    items: list[BundleItemInput]


class BundleUpdateRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    price: float | None = None
    active: bool | None = None
    items: list[BundleItemInput] | None = None
